frontend/src/metabase/databases/components/DatabaseForm/DatabaseForm.tsx

```diff
@@ -102,6 +106,7 @@ interface DatabaseFormBodyProps {
   isCachingEnabled: boolean;
   onEngineChange: (engineKey: string | undefined) => void;
   onCancel?: () => void;
+  onFormDirty?: (isDirty: boolean) => void;
```

Comment:
it's always better to pass config as an object, not as boolean, so `onFormDirty{{isDirty: true})` is more readable than `onFormDirty(true)`, imagine, you extend this method with another param, so it'll become `onFormDirty(true, true, 'yes')`